### PR TITLE
Added access for files

### PR DESCRIPTION
**Pull request by VLiamin**

### CODE REVIEW

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs [hunk 1]
```diff
@@ -33,7 +33,6 @@ private void AddProjectsTable(MigrationBuilder migrationBuilder)
         constraints: table =>
         {
           table.PrimaryKey("PK_Projects", x => x.Id);
-          table.UniqueConstraint("UX_Project_Name_Unique", x => x.Name);
```
Comment (MariyaSin) [inline src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs:36]: why do you remove it?

Comment (VLiamin) [reply src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs]: Done

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 2]
```diff
@@ -423,7 +434,26 @@ public async Task<OperationResultResponse<ProjectResponse>> ExecuteAsync(GetProj
         department = (await GetDepartmentAsync(dbProject.Id, null, response.Errors))?.FirstOrDefault();
       }
 
-      List<Guid> files = dbProject.Files.Select(x => x.FileId).ToList();
+      AccessType accessType = AccessType.Public;
+
+      DbProjectUser user = null;
+      List<DbProjectUser> users = await _userRepository.GetAsync(new List<Guid>() { _httpContextAccessor.HttpContext.GetUserId() });
+      if (users.Count > 0)
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:441]: use Any()

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 3]
```diff
@@ -423,7 +434,26 @@ public async Task<OperationResultResponse<ProjectResponse>> ExecuteAsync(GetProj
         department = (await GetDepartmentAsync(dbProject.Id, null, response.Errors))?.FirstOrDefault();
       }
 
-      List<Guid> files = dbProject.Files.Select(x => x.FileId).ToList();
+      AccessType accessType = AccessType.Public;
+
+      DbProjectUser user = null;
+      List<DbProjectUser> users = await _userRepository.GetAsync(new List<Guid>() { _httpContextAccessor.HttpContext.GetUserId() });
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:440]: add FirstOrDeafult() in the end

Comment (MariyaSin) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:440]: if ( dbProjectUser is not null)
{
      accessType = dbProjectUser.Role == ProjectRoleType.Manager? AccessType.Manager: AccessType.Team;

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 4]
```diff
@@ -423,7 +434,26 @@ public async Task<OperationResultResponse<ProjectResponse>> ExecuteAsync(GetProj
...
+      if (users.Count > 0)
+      {
+        user = users[0];
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:443]: better use firstOrDefault

Comment (VLiamin) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs]: done

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs (hunk 2 above)
Comment (VLiamin) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs]: done

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs (hunk 3 above)
Comment (VLiamin) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs]: done

**Review by MariyaSin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

File: src/ProjectService.Broker/Requests/ImageService.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    public async Task<bool> RemoveImages(List<Guid> imagesIds, List<string> errors)
+    {
+      return imagesIds is null || imagesIds.Any()
+        ? false
+        : (await RequestHandler
+          .ProcessRequest<IRemoveImagesRequest, bool>(
+            _rcRemoveImages,
+            IRemoveImagesRequest.CreateObj(imagesIds: imagesIds, imageSource: ImageSource.User)),
```
Comment (MariyaSin) [inline src/ProjectService.Broker/Requests/ImageService.cs:104]: ImageSourse.Project

File: src/ProjectService.Broker/Requests/ImageService.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            errors,
+            _logger).Item1;
```
Comment (MariyaSin) [inline src/ProjectService.Broker/Requests/ImageService.cs:106]: exception is something wrong, _ogger)?.Body

File: src/ProjectService.Broker/Requests/Interfaces/IImageService.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ProjectService.Broker.Requests.Interfaces
+{
+  [AutoInject]
+  public interface IImageService
+  {
+    Task<List<ImageInfo>> GetImagesAsync(List<Guid> imagesIds, ImageSource imageSource, List<string> errors);
```
Comment (MariyaSin) [inline src/ProjectService.Broker/Requests/Interfaces/IImageService.cs:14]: You in image service => GetAsync

File: src/ProjectService.Broker/Requests/Interfaces/IImageService.cs [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+    Task<List<Guid>> CreateImageAsync(List<ImageContent> projectImages, List<string> errors);
```
Comment (MariyaSin) [inline src/ProjectService.Broker/Requests/Interfaces/IImageService.cs:16]: same

File: src/ProjectService.Broker/Requests/Interfaces/IPositionService.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ProjectService.Broker.Requests.Interfaces
+{
+  [AutoInject]
+  public interface IPositionService
+  {
+    Task<List<PositionData>> GetPositionsAsync(List<Guid> usersIds, List<string> errors);
```
Comment (MariyaSin) [inline src/ProjectService.Broker/Requests/Interfaces/IPositionService.cs:12]: same

File: src/ProjectService.Broker/Requests/UserService.cs [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    {
+      if (projectUsers == null || !projectUsers.Any())
+      {
+        return null;
+      }
+
+      List<Guid> usersIds = projectUsers.Select(x => x.UserId).ToList();
+
+      List<UserData> usersFromCache = await _globalCache.GetAsync<List<UserData>>(Cache.Users, usersIds.GetRedisCacheHashCode());
+
```
Comment (MariyaSin) [inline src/ProjectService.Broker/Requests/UserService.cs:49]: remove empy line

File: src/ProjectService.Broker/Requests/ImageService.cs (hunk 6 above)
Comment (VLiamin) [reply src/ProjectService.Broker/Requests/ImageService.cs:91]: done

**Pull request merged**